Metalama.Framework.Engine/Linking/LinkerInjectionStep.TransformationCollection.cs

```diff
@@ -164,6 +166,16 @@ public void AddAutoPropertyWithSynthesizedSetter( PropertyDeclarationSyntax decl
             }
         }
 
+        public void AddAutoPropertyWithSynthesizedSetter( IPropertyBuilder property )
+        {
+            Invariant.Assert( property is { IsAutoPropertyOrField: true, Writeability: Writeability.ConstructorOnly } );
+
+            lock ( this._autoPropertyWithSynthesizedSetterSyntax )
+            {
+                this._autoPropertyWithSynthesizedSetterBuilders.Add( property );
+            }
```

Comment:
This is locking on a different collection than the one that's being added to. Is that intentional?